### ISSUE TITLE
Sequel support

### ISSUE THREAD

**Issue by gencer** [scoutapp/scout_apm_ruby#95]
Maybe we can examine queries executed by Sequel? RackMini supports Sequel and easily handle it. Maybe you can see the source code and adapt to ScoutApp?

After I GoLive, In production, I would like and love to see how much costs on Sequel operations by using Scout. Otherwise, I just not gonna see it and do not know how much time spent or which query is slow by looking to, ehrm, nothing.

**Comment by matthewford** [comment]
I would like to see this

**Comment by ioquatix** [comment]
Rather than "Sequel" or "ActiveRecord", why not just instrument the underling SQL libraries? Then it will work with everything.

**Comment by matthewford** [comment]
@ioquatix Pretty sure Sequel is the underlying library though 🤔

**Comment by ioquatix** [comment]
Doesn't it use the `pg`, `mysql2` or related gem for actually sending queries to the database?

**Comment by gencer** [comment]
@ioquatix yes it does.

**Comment by ioquatix** [comment]
I guess the challenge is if we instrument the raw query, you might have trouble putting it back in the context of the higher level operation, e.g. joins, transactions, `includes`, etc.

**Comment by gencer** [comment]
I agree. However, At least a basic support for Sequel should be in. Currently, the only reason I do not use DevTrace is that I cannot see how long it takes my queries. RackMini, S**light and Peek can see my queries ans show it to me. If they can do, Why not Scout?

Especially giving credit to DevTrace being an awesome development tool both as a profiling and **UI**. I personally hate RackMini because it is so ugly.

At least, Instrumenting RAW Queries is a start.

**Comment by alecgorge** [comment]
Perhaps we could use Skylight's [sequel probe](https://github.com/skylightio/skylight-ruby/blob/master/skylight-core/lib/skylight/core/probes/sequel.rb) as a jumping off point?

**Comment by cschneid** [comment]
Sorry, but we can't use or look at Skylight's code due to licensing.

**Comment by cschneid** [comment]
I will be looking at Sequel support soon though, I think basic instrumentation shouldn't be too hard.

**Comment by gencer** [comment]
@cschneid, any news on this?

**Comment by cschneid** [comment]
I've started looking at Sequel, but haven't formalized it into a proper instrument yet.  I'll see if I can find time this week to get it released.

**Comment by gencer** [comment]
@cschneid Thank you! I will be very happy to test your branch/PR and give feedback about it. Just ping me if you need anything.

**Comment by soumyaray** [comment]
Really looking forward to this. As a bonus, ROM-RB also uses Sequel as its underlying databasing library, so that whole ecosystem might also get supported automatically!

**Comment by gencer** [comment]
@cschneid, how is going on there? Any progress? Just pinging. _Sorry if I disturb you_.

**Comment by gencer** [comment]
@cschneid Any progress?
